protocol/signalfx/datapoint.go

```diff
@@ -262,3 +264,70 @@ func SetupJSONV1Paths(r *mux.Router, handler http.Handler) {
 	SetupJSONByPaths(r, handler, "/datapoint")
 	SetupJSONByPaths(r, handler, "/v1/datapoint")
 }
+
+// ErrorTrackerHandler behaves like a http handler, but tracks error returns from a ErrorReader
+type ErrorTrackerHandler struct {
+	TotalErrors int64
+	reader      ErrorReader
+	Logger      log.Logger
```

Comment:
Does the logger need to be public?

Reply:
Probably now, but I'd keep the original interaface